src/test/java/testsuite/simple/AwsIamAuthenticationHelperTest.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation and/or
+ * other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package testsuite.simple;
+
+import com.mysql.cj.exceptions.CJException;
+import com.mysql.cj.protocol.a.authentication.AwsIamAuthenticationTokenHelper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+@TestMethodOrder(MethodOrderer.Alphanumeric.class)
+public class AwsIamAuthenticationHelperTest {
+
+    private static final int PORT = 3306;
+
+    /**
+     *  Attempt to create new AWSIAMAuthenticationHelper with valid host name and region
+     */
+    @Test
+    public void test_1_ValidHostAndRegion() {
+        Assertions.assertNotNull(new AwsIamAuthenticationTokenHelper("MyDBInstanceName.SomeServerName.us-east-2.rds.amazonaws.com", PORT));
+    }
+
+    /**
+     *  Attempt to create new AWSIAMAuthenticationHelper with invalid RDS format
+     */
+    @Test
+    public void test_2_NotRdsHost() {
+        Assertions.assertThrows(
+            CJException.class,
+            () -> new AwsIamAuthenticationTokenHelper(
+                "MyDBInstanceName.SomeServerName.us-east-2.notRDS.amazonaws.com", PORT));
+    }
+
+    /**
+     *  Attempt to create new AWSIAMAuthenticationHelper with invalid hostname
+     */
+    @Test
+    public void test_3_NotAmazonHost() {
+        Assertions.assertThrows(
+            CJException.class,
+            () -> new AwsIamAuthenticationTokenHelper("MyDBInstanceName.SomeServerName.us-east-2.rds.notAmazon.com", PORT));
+    }
+
+    /**
+     *  Attempt to create new AWSIAMAuthenticationHelper with empty hostname
+     */
+    @Test
+    public void test_4_EmptyHost() {
+        Assertions.assertThrows(
+            CJException.class,
+            () -> new AwsIamAuthenticationTokenHelper("MyDBInstanceName.SomeServerName.us-east-2.rds. .com", PORT));
+    }
+
+    /**
+     *  Attempt to create new AWSIAMAuthenticationHelper with IP Address as hostname
```

Comment:
You can use tags in your Javadoc.
Specifically, when referring to classes you can use `{@link AwsIamAuthenticationTokenHelper}`.
1. it helps you spot typos
2. it creates a hyperlink to the Java class, pretty convenient I'd say
![image](https://user-images.githubusercontent.com/64801825/134992185-bb921c87-b360-4995-bc2e-847bcf4bd6b7.png)

```suggestion
     *  Attempt to create new {@link AwsIamAuthenticationTokenHelper} with invalid RDS format
```

Please use full stops in your comments.